### PR TITLE
test: automate icon component testing

### LINKED ISSUE #1529: chore(testing): Consolidate individual icon tests into a single auto-discovery test
### Current Behavior

We currently have **8 separate test files** for icon components under `src/__testing__/`:

- `AddIcon.test.tsx`
- `ApplicationIcon.test.tsx`
- `BusIcon.test.tsx`
- `CircleIcon.test.tsx`
- `CodeIcon.test.tsx`
- `PeopleIcon.test.tsx`
- `ReplyIcon.test.tsx`
- `TableChartIcon.test.tsx`

Each file follows the exact same pattern — render the icon and assert `width`/`height`. This means:

1. Only **8 out of 230+** icon components have test coverage.
2. Every new icon requires manually creating a new test file.
3. There's a lot of duplicated boilerplate across files.
---

**[Contributor Guide](https://docs.meshery.io/project/contributing)**

- 🎨 Wireframes and [designs for Sistent site](https://www.figma.com/file/ekm0Hh6M25bSKpP5BcIJom/Sistent-Design-System?type=design&node-id=596%3A5227&mode=design&t=8MExPykHN1vQFCTZ-1) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/GvB8SudhEOoq3JOvoLaoMs)

### PR DESCRIPTION
**Pull request by rishiraj38**
**Notes for Reviewers**

This PR fixes #1529 
Replaces redundant, manual test files for individual icons with a single, automated test suite (`Icons.test.tsx`). 

This new test dynamically iterates over all icon components exported from the `src/icons/index.ts` barrel file and verifies that they:
- Render without crashing
- Produce an `<svg>` element
- Properly forward `width` and `height` props

**Benefits:**
- **Zero Maintenance**: New icons added to the barrel file are automatically tested. No manual test file creation needed.
- **Coverage**: Scales coverage from 8 hardcoded icons to all 230+ current icons instantly.

Screenshot 2026-05-11 at 10 35 13 PM

**[Signed commits](../blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**

- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by rishiraj38** [comment]
@Rajesh-Nagarajan-11 
 while running the tests I'm getting some React DOM property warnings from a few icon source files. They use HTML SVG attributes like clip-path, stroke-linecap, stroke-linejoin, fill-rule, clip-rule, fill-opacity, stroke-width instead of React's camelCase equivalents (clipPath, strokeLinecap, etc.).

All tests pass fine, it's just noisy console output. Should I fix those in the icon files as part of this PR or keep it separate?

File: src/__testing__/Icons.test.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import * as AllIcons from '../icons';
+import { SistentThemeProviderWithoutBaseLine } from '../theme';
+
+// grab all icon components from the barrel file
+// types get stripped at runtime so we only get actual components here
+// also checks for uppercase first letter to follow React component naming convention
+const iconEntries = Object.entries(AllIcons).filter(
+  ([name, value]) => typeof value === 'function' && /^[A-Z]/.test(name)
+) as [string, React.ComponentType<Record<string, unknown>>][];
+
```
[the remark this answers is not in the document]
Comment (rishiraj38) [reply src/__testing__/Icons.test.tsx:12]: Good point on FilledCircleIcon being missed! However, rather than complicating the entire test suite logic for one edge case, I will just update FilledCircleIcon to export as a standard component like the rest. Much cleaner this way!

File: src/__testing__/Icons.test.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+describe.each(iconEntries)('%s', (_name, IconComponent) => {
+  it('renders an <svg> element and forwards width/height props', () => {
+    const { container } = renderWithTheme(<IconComponent width={32} height={32} />);
+    const svg = container.querySelector('svg');
+
+    expect(svg).not.toBeNull();
+    expect(svg!.getAttribute('width')).toBe('32');
+    expect(svg!.getAttribute('height')).toBe('32');
```
[the remark this answers is not in the document]
Comment (rishiraj38) [reply src/__testing__/Icons.test.tsx]: Great catch! I applied this and it immediately caught a bug where EditIcon had its width and height props swapped. I've updated the test and fixed the icon.

**Comment by Rajesh-Nagarajan-11** [comment]
Lint fails

**Comment by rishiraj38** [comment]
> Lint fails

After I applied copilot's 1st suggestion it immediately caught a bug where EditIcon had its width and height props swapped.

**Comment by rishiraj38** [comment]
All tests are passing now after this [PR](https://github.com/layer5io/sistent/pull/1544) merger. However, it still has a few invalid DOM warnings.  Next, the target is to solve those.

**Review by Rajesh-Nagarajan-11 — APPROVED** [review]
